File: ts-src/core/exceptions.ts
/**
 * Exception hierarchy for LambdaLLM.
 */

export class LambdaLLMError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'LambdaLLMError';
  }
}

export class ModelInvocationError extends LambdaLLMError {
  public retryable: boolean;

  constructor(message: string, retryable = true) {
    super(message);
    this.name = 'ModelInvocationError';
    this.retryable = retryable;
  }
}

export class TimeoutError extends LambdaLLMError {
  constructor(message: string) {
    super(message);
    this.name = 'TimeoutError';
  }
}

export class BudgetExceededError extends LambdaLLMError {
  constructor(message: string) {
    super(message);
    this.name = 'BudgetExceededError';
  }
}

export class ConfigurationError extends LambdaLLMError {
  constructor(message: string) {
    super(message);
    this.name = 'ConfigurationError';
  }
}

File: src/lambdallm/core/exceptions.py
"""LambdaLLM exception hierarchy."""


class LambdaLLMError(Exception):
    """Base exception for all LambdaLLM errors."""

    pass


class ModelInvocationError(LambdaLLMError):
    """Raised when a model invocation fails."""

    def __init__(self, message: str, retryable: bool = True):
        super().__init__(message)
        self.retryable = retryable


class TimeoutError(LambdaLLMError):
    """Raised when approaching Lambda timeout."""

    pass


class BudgetExceededError(LambdaLLMError):
    """Raised when cost budget is exceeded."""

    pass


class ConfigurationError(LambdaLLMError):
    """Raised when framework configuration is invalid."""

    pass


class ProviderError(LambdaLLMError):
    """Raised when a model provider encounters an error."""

    pass


class StateError(LambdaLLMError):
    """Raised when state management fails."""

    pass
